include/alpaka/meta/NdLoop.hpp

```diff
@@ -123,13 +124,13 @@ namespace alpaka
                         dim::Dim<TIndex>::value == dim::Dim<TExtentVec>::value,
                         "The dimensions of the iteration vector and the extent vector have to be identical!");
                     static_assert(
-                        dim::Dim<TIndex>::value > Tdim,
+                        dim::Dim<TIndex>::value > Tdim0,
```

Comment:
I am a bit confused with the `Tdim0`s.
Shouldn't this always be `TdimN` of the current index?

Reply:
As we are executing the loops given the indices from left to right `Tdim0` is always the current dimension/index. All the others are forwarded to the next recursive invocation where `Tdim1` is now the new `Tdim0`.

Reply:
I think the code is correct: `Tdim0` is always for the current index, since we iterate over that and so the current `Tdim1` will become `Tdim0` next time, etc.